superset-websocket/index.js

```diff
@@ -0,0 +1,175 @@
+const config = require('./config.json');
+const http = require('http');
+const WebSocket = require('ws');
+const jwt = require('jsonwebtoken');
+const cookie = require('cookie');
+const Redis = require('ioredis');
+const redis = new Redis(config.redis);
+
+const server = http.createServer();
+const wss = new WebSocket.Server({ noServer: true });
+
+const globalEventStreamName = `${config.streamPrefix}full`;
+let channels = {};
+let lastFirehoseId = '$';
+let redisRequestCount = 0;
+
+
+/*
+NOT using Redis Stream consumer groups due to the fact that they only read
+a subset of the data for a stream, and Websocket clients have a persistent
+connection to each app instance, requiring access to all data in a stream.
+Horizontal scaling of the websocket app means having multiple websocket servers,
+each with full access to the Redis Stream.
+
+Stream options:
```

Comment:
which of these options do you recommend here?

Reply:
Option 3 was originally selected after performance testing each, though this has changed a bit due to the implications of public dashboards, in which the user has no ID value. I'll update this.